openlibrary/catalog/get_ia.py

```diff
@@ -196,49 +129,59 @@ def get_data(loc):
     return buf
 
 def get_from_archive(locator):
+    """
+    Gets a single binary MARC record from within an Archive.org
+    bulk MARC item - data only.
+
+    :param str locator: Locator ocaid/filename:offset:length
+    :rtype: str|None
+    :return: Binary MARC data
+    """
+    data, offset, length = get_from_archive_bulk(locator)
+    return data
+
+def get_from_archive_bulk(locator):
+    """
+    Gets a single binary MARC record from within an Archive.org
+    bulk MARC item, and return the offset and length of the next
+    item.
+    If offset or length are `None`, then there is no next record.
+
+    :param str locator: Locator ocaid/filename:offset:length
+    :rtype: (str|None, int|None, int|None)
+    :return: (Binary MARC data, Next record offset, Next record length)
+    """
```

Comment:
error handling? What if last record in stream?

Reply:
It should return (data, None, None) if it is the last record, and (None, None, None) if seeking beyond, without blowing up. I will confirm, and see if there is a neat way to add a test to prove this without making live requests.